components/channel_header/channel_header.jsx

```diff
@@ -499,17 +491,18 @@ export default class ChannelHeader extends React.Component {
                     key='add_members'
                     role='presentation'
                 >
-                    <button
-                        className='style--none'
+                    <ToggleModalButtonRedux
                         id='channelAddMembersGroup'
                         role='menuitem'
-                        onClick={this.openDirectMessageModal}
+                        modalId={ModalIdentifiers.CREATE_DM_CHANNEL}
+                        dialogType={MoreDirectChannels}
+                        dialogProps={{isExistingChannel: true}}
                     >
                         <FormattedMessage
                             id='channel_header.addMembers'
                             defaultMessage='Add Members'
                         />
-                    </button>
+                    </ToggleModalButtonRedux>
                 </li>
```

Comment:
This is the actual fix.
Previously, the button onClick function dispatches an action type of TOGGLE_DM_MODAL. Unfortunately, the listener for that was recently removed.
Instead of reverting such change, this PR converts the use of this modal into the available redux-style views.